src/js/storage/storage.js

```diff
@@ -1,31 +1,42 @@
 /**
- * Class to store items in localstorage.
- * This code I got from Oliver
- * Joakim Tveter changed the third parameter nameing for clarity
+ * @class Store
+ * @classdesc A class for storing data in the browser.
+ * @param {string} key - The key to store the data under.
+ * @param {object} state - The data to store.
+ * @param {boolean} [persist=false] - Whether or not to persist the data. Changes the storage from session to local.
  */
 export class Store {
-  constructor(key, state, session = false) {
-    this.storage = session ? sessionStorage : localStorage;
-
+  constructor(key, state, persist = false) {
```

Comment:
Is there a reason why you prefer session storage? 
I am trying to get role in localStorage but can't now because of this change.

Reply:
Hi, I thought the default behaviour is more secure if `persist = false`. If you accidentally forget to set it, at least you're not saving user data without permission. 

If you want to save something to localStorage instead, just use new Store("yourKey", state, true) you save it.

Reply:
But doesn't that mean we will override state each time we create a new Store?

Reply:
> But doesn't that mean we will override state each time we create a new Store?

Each Store instance is independent and will not override other instances, provided that you use a unique key for each. When you instantiate Store with a new key, it will manage its own state in localStorage or sessionStorage without affecting others. Just ensure that the keys you use are distinct for the different data sets you want to store. 


To access the same `Store` instance from several different files, you have a couple of options. One is to create a "singleton" of the store, ensuring that the same instance is used wherever it's imported. The other option is to create a centralised store management system where you instantiate and manage all your stores.

Here's how you can create a singleton of your `Store`:

1. **Singleton Pattern:**
   You can create a file that exports an instance of the `Store`. This instance will be created the first time the file is imported, and subsequent imports will use the same instance.

   ```js
   // storeSingleton.js
   import { Store } from './src/js/storage/storage.js';

   // Create a singleton instance of Store
   export const myStoreSingleton = new Store('myUniqueKey', {}, true);
   ```

   Then, in other files, you can import `myStoreSingleton`:

   ```js
   // otherFile.js
   import { myStoreSingleton } from './storeSingleton';

   // Use the singleton instance
   const state = myStoreSingleton.state;
   ```

2. **Centralised Store Management:**
   Another approach is to have a central file where you instantiate and export all your stores. This is particularly useful if you have multiple stores but want to manage them centrally.

   ```js
   // ./centralStore.js
   import { Store } from './src/js/storage/storage.js';

   export const userStore = new Store('userKey', {}, true);
   export const settingsStore = new Store('settingsKey', {}, true);
   // ... other stores
   ```

   You can then import these stores in different files:

   ```js
   // usageFile.js
   import { userStore, settingsStore } from './centralStore';

   // Access the stores as needed
   const userState = userStore.state;
   const settingsState = settingsStore.state;
   ```

Both methods ensure that you're working with the same instance(s) of `Store` across different parts of your application, maintaining consistency in your data handling.